app/src/main/java/com/example/teamyuml/decentworkmobile/fragments/ListViewFragment.java

```diff
@@ -0,0 +1,216 @@
+import com.example.teamyuml.decentworkmobile.VolleyInstance;
+import com.example.teamyuml.decentworkmobile.views.CustomListView;
+import com.example.teamyuml.decentworkmobile.views.CustomWorkerView;
+import com.example.teamyuml.decentworkmobile.views.NoticeDetails;
+import com.example.teamyuml.decentworkmobile.views.WorkerDetails;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+public class ListViewFragment extends Fragment {
+    private ArrayList<HashMap<String, String>> data;
+    private ListView listView;
+    private String URL;
+    private int listViewId;
+    private int listLayoutId;
+    private Method callMethod;
+    private ArrayAdapter<HashMap<String, String>> adapterClass;
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+    }
+
+    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle bundle) {
+        Bundle fields = getArguments();
+        data = new ArrayList<>();
+
+        try {
+            if (fields != null) {
+                initializeFragment(fields);
+                callMethod.invoke(this);
+            }
+        } catch (NoSuchMethodException | InvocationTargetException | IllegalAccessException e) {
+            Toast.makeText(
+                getActivity() ,"Brak dostępu do metody.", Toast.LENGTH_LONG).show();
+            e.printStackTrace();
+        }
+
+        View v = inflater.inflate(listLayoutId, container, false);
+        listView = v.findViewById(listViewId);
+        return v;
+    }
+
+    /**
+     * Initialize fragment fields.
+     * @param bundle - Bundle from {@link Fragment#getArguments()} method
+     */
+    private void initializeFragment(Bundle bundle) throws NoSuchMethodException {
+        URL = bundle.getString("url");
+        listViewId = bundle.getInt("listViewId");
+        listLayoutId = bundle.getInt("listLayoutId");
+        String methodName = bundle.getString("methodName");
+
+        if (methodName != null) {
+            callMethod = getClass().getDeclaredMethod(methodName);
+            callMethod.setAccessible(true);
+        } else {
+            throw new NoSuchMethodException();
+        }
+    }
+
+    /**
+     * Set adapter to ListView and add clickListener event to get clicked data
+     */
+    private void initListView() {
+        listView.setAdapter(adapterClass);
+
+        listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+            public void onItemClick(AdapterView<?> parent, View v, int position, long id) {
+                String clickedItem = data.get(position).get("id").toString();
+                System.out.println(listLayoutId);
+
+                if(listLayoutId == 2131361828) {
```

Comment:
I don't think this is good idea tbh. What if we have to use this fragment with next four other lists? IMO we can add new String field like className set it to Bundle in NoticeList like I set method name and then here use Class.forName(name) insted of e.g NoticeDetails.class. The only thing which I don't know is if we need to use package name or not.